tests/elements-regression/src/pages/editor-page.js

```diff
@@ -110,27 +125,27 @@ module.exports = class EditorPage extends BasePage {
 	}
 
 	async waitForElementRender( id ) {
-		let isLoading;
-
-		try {
-			await this.getPreviewFrame().waitForSelector(
-				`.elementor-element-${ id }.elementor-loading`,
-				{ timeout: 500 },
-			);
-
-			isLoading = true;
-		} catch ( e ) {
-			isLoading = false;
+		const sleep = ( ms ) => new Promise( ( resolve ) => setTimeout( resolve, ms ) );
+		let elements = await this.getPreviewFrame().evaluate( 'window.readyElements' );
+		let i = 0;
+
+		while ( ! elements.includes( id ) && i < 10 ) {
+			i++;
+			await sleep( 100 );
+			elements = await this.getPreviewFrame().evaluate( 'window.readyElements' );
 		}
 
-		if ( isLoading ) {
-			await this.getPreviewFrame().waitForSelector(
-				`.elementor-element-${ id }.elementor-loading`,
-				{ state: 'detached' },
-			);
+		// Add extra sleep to make sure the element is fully loaded.
+		await sleep( 200 );
```

Comment:
Can we get rid of this? This function makes us sleep for at least 300ms - seems like a lot
```suggestion
		await this.page.waitForTimeout( 200 );
```